### DISCUSSION TITLE
[v0.4] High-level interface for Mixture of Agents

### DISCUSSION THREAD

**Discussion by MarkYav** [microsoft/autogen#4234] [Q&A]
Thank you a lot for the very detailed docs regarding [Mixture of Agents](https://microsoft.github.io/autogen/dev/user-guide/core-user-guide/design-patterns/mixture-of-agents.html) in the `autogen-core` package!

Do we (or will we) have a high-level API for this in the `autogen-agentchat` package (like `RoundRobinGroupChat`)?

**Comment by ekzhu** [comment]
This could be big. Though we are still waiting to see some solid application scenarios. Before that, anyone is welcome to create a community package that extends agentchat with a mixture of agents team. 

See community packages: https://microsoft.github.io/autogen/dev/user-guide/extensions-user-guide/index.html

We can share and index.

**Reply by MarkYav** [reply]
@ekzhu Is there a guide on how to make a nested chat for AutoGen v0.4?

**Reply by ekzhu** [reply]
The official design for nested chat is very much in flight at the moment. You can take a look at the implementation of SocietyOfMind agent: https://microsoft.github.io/autogen/dev/reference/python/autogen_agentchat/autogen_agentchat.agents.html#autogen_agentchat.agents.SocietyOfMindAgent

The general idea is to put a team inside an agent. You can also implement a different agent with a nested team.
